src/rdrand.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2018 Developers of the Rand project.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Implementation for SGX using RDRAND instruction
+use crate::Error;
+use core::mem;
+use core::arch::x86_64::_rdrand64_step;
+use core::num::NonZeroU32;
+
+#[cfg(not(target_feature = "rdrand"))]
+compile_error!("enable rdrand target feature!");
+
+// Recommendation from "Intel® Digital Random Number Generator (DRNG) Software
+// Implementation Guide" - Section 5.2.1 and "Intel® 64 and IA-32 Architectures
+// Software Developer’s Manual" - Volume 1 - Section 7.3.17.1.
+const RETRY_LIMIT: usize = 10;
+const WORD_SIZE: usize = mem::size_of::<usize>();
```

Comment:
Why do you use word size instead of `u64` size? `_rdrand64_step` explicitly works with `u64` and not with `usize`. Yes, `x86_64-fortanix-unknown-sgx` is a 64-bit target, but I think it will be better to use `u64` here.

Reply:
Whoops, I accidentally committed some of my changes that will make this work on 32-bit x86, should be fixed now.